### ISSUE TITLE
Support for publicNetworkAccess property for Microsoft.KeyVault

### ISSUE THREAD

**Issue by ChadFisher-Zywave** [hashicorp/terraform-provider-azurerm#17533]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Support publicNetworkAccess property on key vaults, introduced in API version 2021-10-01. Could function similar to `public_network_access_enabled` property already present on other resource types. 

### New or Affected Resource(s)/Data Source(s)

azurerm_key_vault

### Potential Terraform Configuration

```hcl
resource "azurerm_key_vault" "example" {
  name                          = "examplekeyvault"
  location                      = azurerm_resource_group.example.location
  resource_group_name           = azurerm_resource_group.example.name
  tenant_id                     = data.azurerm_client_config.current.tenant_id
  sku_name                      = "standard"
  public_network_access_enabled = false
}
```

### References

https://docs.microsoft.com/en-us/azure/templates/microsoft.keyvault/2021-10-01/vaults

**Comment by sinbai** [comment]
Hi @ChadFisher-Zywave,

Thanks for raising this issue here. Support for publicNetworkAccess property for Microsoft.KeyVault has been released in [v3.19.0 of the Terraform Provider](https://github.com/hashicorp/terraform-provider-azurerm/blob/v3.19.0/CHANGELOG.md). Please let me know if you need anything else.

In addition,  if it works for you, are you okey we close this issue?

**Comment by sinbai** [comment]
@ChadFisher-Zywave since the issue has been fixed in [v3.19.0 of the Terraform Provider](https://github.com/hashicorp/terraform-provider-azurerm/blob/v3.19.0/CHANGELOG.md), are you okey we close this issue?

**Comment by mybayern1974** [comment]
Per sinbai's suggestion, I'm closing this issue. Feel free to call out this issue needs to be re-activated.
